stacks/mlflow/tracking_merlion.py

```diff
@@ -0,0 +1,168 @@
+"""
+Use MLflow to track the events and outcome of an ML experiment program using Merlion.
+"""
+import os
+
+import numpy as np
+import pandas as pd
+from crate import client
+from crate.client.exceptions import ProgrammingError
+from merlion.evaluate.anomaly import TSADMetric
+from merlion.models.defaults import DefaultDetector, DefaultDetectorConfig
+from merlion.utils import TimeSeries
+
+import mlflow
+
+
+def connect_database():
+    """
+    Connect to CrateDB, and return database connection object.
+    """
+    dburi = os.getenv("CRATEDB_HTTP_URL", "http://crate@localhost:4200")
+    return client.connect(dburi)
+
+
+def provision_data():
+    """
+    Download Numenta Anomaly Benchmark data, and load into database.
+    """
+    data = pd.read_csv(
+        "https://raw.githubusercontent.com/numenta/NAB/master/data/realKnownCause/machine_temperature_system_failure.csv"
+    )
+
+    # Connecting to crate
+    conn = connect_database()
+    try:
+        conn.cursor().execute("SELECT * FROM machine_data LIMIT 1;")
+        return
+    except ProgrammingError as ex:
+        if "Relation 'machine_data' unknown" not in str(ex):
+            raise
```

Comment:
Minor: Looking at the related system classes to check table existence would be safer than expecting a concrete exception:
```suggestion
    cursor = conn.cursor()
    cursor.execute("SELECT table_name FROM information_schema.tables WHERE table_name = 'machine_data'  and table_schema = 'doc'");
    if cursor.rowcount > 0:
      raise Exception("Relation 'machine_data' already exists")
```

Reply:
Introspecting the information schema tables is definitively a better way to handle this. Thanks!

Reply:
I revisited this detail, and came to a different conclusion, see https://github.com/crate-workbench/mlflow-cratedb/pull/13#discussion_r1320781162.